tooling/minitest/src/pass/concurrency.rs

```diff
@@ -0,0 +1,71 @@
+use crate::*;
+
+#[test]
+fn arbitrary_order() {
+
+    // A function that writes x to the global(1).
+    fn write(x: u32) -> Function {
+        let locals = [<u32>::get_ptype()];
+
+        let b0 = block!(
+            storage_live(0),
+            assign(local(0), load(global::<u32>(0))),
+            acquire(load(local(0)), 1)
+        );
+        let b1 = block!(
+            assign(global::<u32>(1), const_int::<u32>(x)),
+            release(load(local(0)), 2)
+        );
+        let b2 = block!(return_());
+
+        function(Ret::No, 0, &locals, &[b0, b1, b2])
+    }
+
+    // Main function, creates a lock and two threads.
+    let locals = [<u32>::get_ptype(), <u32>::get_ptype()];
+
+    let b0 = block!(
+        create_lock(global::<u32>(0), 1)
+    );
+    let b1 = block!(
+        storage_live(0),
+        spawn(fn_ptr(1), Some(local(0)), 2)
+    );
+    let b2 = block!(
+        storage_live(1),
+        spawn(fn_ptr(2), Some(local(1)), 3)
```

Comment:
There's a lot of bare numbers here. Would be good to explain what they mean in comments.
(Would be even better if they could be handled automatically by the builder API but that is a bigger change.^^)

What are the fn_ptr numbers here?

Reply:
Yes I already thought about a better way to write programs. But I came to the conclusion minimize would be the optimal solution here.

The numbers stand for the function name. It is given based on the position in the function slice. Will comment this.

Reply:
There might be something nice even without going all the way to MIR... if you have time left after finishing the concurrency stuff we could talk about that. For now some comments will do.